core/src/main/java/org/mule/runtime/core/routing/UntilSuccessful.java

```diff
@@ -87,9 +87,7 @@ public void initialise() throws InitialisationException {
     this.untilSuccessfulStrategy.setUntilSuccessfulConfiguration(this);
 
     initialiseIfNeeded(untilSuccessfulStrategy, muleContext);
-    String flowName = flowConstruct.getName();
-    String clusterId = muleContext.getClusterId();
-    eventKeyPrefix = flowName + "-" + clusterId + "-";
+    eventKeyPrefix = getLocation().getParts().get(0).getPartPath() + "-" + muleContext.getClusterId() + "-";
```

Comment:
Refactor repeated use of this type of thing into static helpers in ComponentLocation.